src/lambda/commands/uploadLambda.ts

```diff
@@ -412,3 +441,113 @@ async function uploadZipBuffer(
         localize('AWS.lambda.upload.done', 'Successfully uploaded Lambda function {0}', lambda.name)
     )
 }
+
+export async function findApplicationJsonFile(startPath: vscode.Uri): Promise<vscode.Uri | undefined> {
+    const isTemplateFile = /template.(yaml|yml|json)$/.test(startPath.fsPath)
+    const parentDir = isTemplateFile ? path.dirname(startPath.fsPath) : startPath.fsPath
+
+    if (isCloud9()) {
+        const found = SystemUtilities.findFile(parentDir, '.application.json')
+        return found ? vscode.Uri.parse(found) : undefined
+    } else {
+        const found = await vscode.workspace.findFiles(
+            new vscode.RelativePattern(parentDir, '**/*.application.json'),
+            undefined,
+            1
+        )
+        return found[0]
+    }
+}
+
+export function getLambdaNamesFromApplicationJson(file: vscode.Uri): string[] | undefined {
+    const appData = JSON.parse(fs.readFileSync(file.fsPath, { encoding: 'utf-8' }).toString())
+    return Object.keys(appData['Functions'])
```

Comment:
1. Doesn't handle invalid JSON. It should at least log. 
2. Doesn't handle unexpected JSON structure (e.g. missing "Functions")
3. Must not fail (return undefined; caller interprets this as "nothing found")